### PR TITLE
Basic Body Setup

### PR DESCRIPTION
**Pull request by esthergassner**
Built the four walls, top corners, bottom bases, flippers (which are still static as of yet), divider wall, and the ball. Code is in major need of refactoring. Also, all diagonal bodies are drawn as lines. Hopefully Ayelet's renderer will be of assistance.

image

### CODE REVIEW

File: src/main/java/pinball/PinballComponent.java [hunk 1]
```diff
@@ -1,16 +1,187 @@
...
+        Body wall = world.createBody(bodyDef);
+
+        FixtureDef fixtureDef = new FixtureDef();
+        PolygonShape shape = new PolygonShape();
+        shape.setAsBox(length, height);
+        fixtureDef.shape = shape;
+        fixtureDef.restitution = 1;
+        wall.createFixture(fixtureDef);
+        return wall;
+    }
```
Comment (HelenSutton) [inline src/main/java/pinball/PinballComponent.java:71]: can just have one method to create lines. create diagonal line can create the walls too

**Comment by esthergassner** [comment]
PinballScreenshot

File: src/main/java/pinball/PinballComponent.java [hunk 2]
```diff
@@ -1,16 +1,187 @@
...
+        topLeftCorner = createDiagonalLine(250f * SCREEN_TO_BOX, 100f * SCREEN_TO_BOX, CORNER_LENGTH * SCREEN_TO_BOX, 150);
+        bottomRightBase = createDiagonalLine(1025 * SCREEN_TO_BOX, 1230f * SCREEN_TO_BOX, BASE_LENGTH * SCREEN_TO_BOX, 145);
+        bottomLeftBase = createDiagonalLine(100f * SCREEN_TO_BOX, 1230f * SCREEN_TO_BOX, BASE_LENGTH * SCREEN_TO_BOX, 35);
+
+        //set up flippers (static for now, but soon to be dynamic and jointed)
+        rightFlipper = createDiagonalLine(735 * SCREEN_TO_BOX, 1435 * SCREEN_TO_BOX, FLIPPER_LENGTH * SCREEN_TO_BOX, 145);
+        leftFlipper = createDiagonalLine(390 * SCREEN_TO_BOX, 1435 * SCREEN_TO_BOX, FLIPPER_LENGTH * SCREEN_TO_BOX, 35);
+
+        //cue the ball
+        ball = createBall(1060 * SCREEN_TO_BOX, 1565 * SCREEN_TO_BOX, radius * SCREEN_TO_BOX);
```
Comment (andrewoid) [inline src/main/java/pinball/PinballComponent.java:54]: Move the SCREEN_TO_BOX multiplication into the create() methods to make this easier to read.

Comment (andrewoid) [inline src/main/java/pinball/PinballComponent.java:54]: Move the pinball body generation into it's own class.

Comment (esthergassner) [reply src/main/java/pinball/PinballComponent.java:54]: This will be fixed in Kaila's PR.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**